### PR TITLE
Fix #163

### LINKED ISSUE #163: .htmlhintrc not used by default
The [usage docs](https://github.com/yaniswang/HTMLHint/wiki/Usage) state that `.htmlhintrc` from the current working directory is used by default.

> search .htmlhintrc file in current directory and all parent directories:
> 
> ```
> htmlhint
> htmlhint test.html
> ```

No default `.htmlhintrc` file is actually used.

### PR DESCRIPTION
**Pull request by phaux**
Make htmlhint check the root for the config file. Fixes #163 

> Same issue here. I added a few `console.log`s and it seems that it quits the loop one iteration too soon and skips checking the root.
>
> So when doing `htmlhint foo/bar/baz/**/*.html` it first figures out the root of the glob which is `foo/bar/baz` and checks `foo/bar/baz`, `foo/bar` and `foo`. It never checks empty string path

### CODE REVIEW

**Comment by trisys3** [comment]
Is this going to get merged anytime soon?

**Comment by phaux** [comment]
My current workaround is to just add `-c .htmlhintrc` to the command, so instead that
```json
{
  "scripts": {
    "test": "htmlhint src/**/*.html"
  }
}
```
I've got
```json
{
  "scripts": {
    "test": "htmlhint -c .htmlhintrc src/**/*.html"
  }
}
```

**Comment by jonsmithers** [comment]
I imagine another workaround could be this:

```bash
npm remove htmlhint
npm install "https://github.com/phaux/HTMLHint.git#patch-1"
```

**Comment by trisys3** [comment]
Are you hinting that @phaux has a patch for this and he will merge it into master soon?

**Comment by jonsmithers** [comment]
@trisys3  I mean....this Pull Request _is_ the patch, but, as you know, it's taking a while to get merged. 
I'm just saying that instead of installing the last official release (which `npm install htmlhint` does), we can install @phaux's patched version by referencing this branch directly (`npm install "https://github.com/phaux/HTMLHint.git#patch-1"`).

**Comment by trisys3** [comment]
Oh, I don't know how GitHub patch URL's work, I'll have to look them up later.

File: bin/htmlhint [hunk 1]
```diff
@@ -300,12 +300,13 @@ function getConfig(configPath, base, formatter){
         if(fs.statSync(base).isDirectory() === false){
             base = path.dirname(base);
         }
-        while(base){
-            var tmpConfigFile = path.resolve(base+path.sep, '.htmlhintrc');
+        while(true){
```
Comment (Shinigami92) [inline bin/htmlhint]: Yes, maybe that works
But can you rewrite this in a way that does not use an always `true` statement in the while clause?

This conflicts with the eslint rule [no-constant-condition](https://eslint.org/docs/rules/no-constant-condition) and we want that later in our project

**Pull request closed without merging**